index.js

```diff
@@ -133,9 +133,9 @@ async function send (ctx, path, opts = {}) {
     }
     ctx.set('Cache-Control', directives.join(','))
   }
-  ctx.type = type(path)
   ctx.body = fs.createReadStream(path)
-
+  ctx.type = ctx.res.type ? ctx.res.type : type(path);
```

Comment:
In summary:

1. If `extension exists`, we need reset `type` to keep consistent with `body`
2. We move `setHeaders`, so we can `set all headers` for special use cases
3. If `no extension`, should we remove `type` (if exists)
  - if so, we just need keep `ctx.type = type(path)`
  - if no, why ?
    * for your use case, if we move `setHeaders`, everything is ok.
 
@nitinbansal1989 any advice ? 😄